src/commonMain/kotlin/fr/acinq/lightning/io/TcpSocket.kt

```diff
@@ -23,12 +23,19 @@ interface TcpSocket {
 
     fun close()
 
-    enum class TLS {
-        SAFE, UNSAFE_CERTIFICATES
+    sealed class TLS {
+        object DISABLED: TLS()
+        object TRUSTED_CERTIFICATES: TLS()
+        object UNSAFE_CERTIFICATES: TLS()
+        data class PINNED_PUBLIC_KEY(
+            // DER-encoded publicKey as base64 string.
+            // (I.e. same as PEM format, without BEGIN/END header/footer)
+            val pubKey: String
+        ): TLS()
```

Comment:
nit: I think it could be nice to add comments here, something like:
- `DISABLED`: used for Lightning connections
- `TRUSTED_CERTIFICATES`: used for Electrum servers when expecting a valid certificate
- `PINNED_PUBLIC_KEY`: used for Electrum servers when expecting a specific public key (for example self-signed certificates)
- `UNSAFE_CERTIFICATES`: only used in tests.

Reply:
Good idea :thumbsup:

Reply:
Done in [1e29485](https://github.com/ACINQ/lightning-kmp/pull/327/commits/1e294851f976629450e1f97a87f124a2a485144c)